### PR TITLE
Move unit tests for  request info

### PR DESCRIPTION
**Pull request by VVeremjova**
Commit was moved from PASA.

In commit  request_info.h and request_info.cc were changed (related to [APPLINK-22306](https://adc.luxoft.com/jira/browse/APPLINK-22306) ).
 Contructor  RequestInfo(RequestPtr request,  const RequestType request_type, conts TimevalStruct start_time, const uint64_t timeout_msec) was deleted as not used
  Contructor  RequestInfo(RequestPtr request,  const RequestType request_type, const uint64_t timeout_msec, const uint32_t app_id, const uint32_t correlation_id) was modified according[ #339](https://github.com/CustomSDL/sdl_panasonic/pull/339/)

Related to [APPLINK-22722](https://adc.luxoft.com/jira/browse/APPLINK-22722)

Please review @OHerasym, @okozlovlux, @LevchenkoS, @VProdanov, @dtrunov, @nk0leg

### CODE REVIEW

File: src/components/application_manager/test/request_controller/request_info_test.cc [hunk 1]
```diff
@@ -167,24 +149,24 @@ TEST_F(RequestInfoTest, AddHMIRequests_RemoveAllRequests) {
...
-            request_info_set_.RemoveByConnectionKey(hmi_connection_key_));
+  EXPECT_EQ(kCountOfRequestsForTest_, request_info_set_.Size());
+  EXPECT_EQ(kCountOfRequestsForTest_,
+            request_info_set_.RemoveByConnectionKey(kHmiConnectionKey_));
   EXPECT_EQ(0u, request_info_set_.Size());
 }
+
 TEST_F(RequestInfoTest, CheckRequestsMaxCount) {
   const uint32_t app_hmi_level_time_scale = 100;
   const uint32_t hmi_level_count = 1000;
```
Comment (okozlovlux) [inline src/components/application_manager/test/request_controller/request_info_test.cc]: @VVeremjova, please add 'u' and check const names

File: src/components/application_manager/test/request_controller/request_info_test.cc [hunk 2]
```diff
@@ -231,92 +210,84 @@ TEST_F(RequestInfoTest, CheckMaxCountOfRequest) {
...
   EXPECT_EQ(2u, request_info_set_.Size());
   EXPECT_EQ(2u, request_info_set_.RemoveMobileRequests());
   EXPECT_EQ(0u, request_info_set_.Size());
 }
 
 TEST_F(RequestInfoTest, AddMobileRequests_RemoveMobileRequestsByConnectionKey) {
-  std::vector<utils::SharedPtr<TestRequestInfo>> requests;
+  std::vector<utils::SharedPtr<RequestInfoForTesting>> requests;
   const uint32_t count_of_mobile_request1 = 200;
   const uint32_t count_of_mobile_request2 = 100;
```
Comment (okozlovlux) [inline src/components/application_manager/test/request_controller/request_info_test.cc]: @VVeremjova, please add 'u' and check const names

File: src/components/application_manager/test/request_controller/request_info_test.cc [hunk 3]
```diff
@@ -395,12 +361,11 @@ TEST_F(RequestInfoTest, RequestInfoSetEqualHash) {
   request_info::RequestInfoSet request_info_set;
   const uint32_t connection_key = 65483;
   const uint32_t corr_id = 65483;
```
Comment (okozlovlux) [inline src/components/application_manager/test/request_controller/request_info_test.cc]: @VVeremjova, please add 'u' and check const names

File: src/components/application_manager/test/request_controller/request_info_test.cc [hunk 4]
```diff
@@ -30,6 +30,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 #include "application_manager/request_info.h"
+#include "application_manager/test_request_info.h"
+#include <stdint.h>
```
Comment (LevchenkoS) [inline src/components/application_manager/test/request_controller/request_info_test.cc]: @VVeremjova, wrong order of includes

File: src/components/application_manager/test/request_controller/request_info_test.cc [hunk 5]
```diff
@@ -325,24 +296,20 @@ TEST_F(RequestInfoTest, AddMobileRequests_RemoveMobileRequestsByConnectionKey) {
...
-                       date_time::DateTime::getCurrentTime(),
-                       i);
+  for (uint32_t i = 0; i < kCountOfRequestsForTest_; ++i) {
+    utils::SharedPtr<RequestInfoForTesting> request = CreateTestInfo(
+        kMobileConnectionKey1_, i, request_info::RequestInfo::HMIRequest, i);
     request_info_set_.Add(request);
   }
 
-  for (uint32_t i = 1; i < count_of_requests_for_test_; ++i) {
+  for (uint32_t i = 1; i < kCountOfRequestsForTest_; ++i) {
```
Comment (LevchenkoS) [inline src/components/application_manager/test/request_controller/request_info_test.cc]: @VVeremjova, why counting starts from 1?

**Comment by LevchenkoS** [comment]
@VVeremjova, reviewed.

File: src/components/application_manager/test/request_controller/request_info_test.cc (hunk 2 above)
Comment (VVeremjova) [inline src/components/application_manager/test/request_controller/request_info_test.cc]: @okozlovlux See d960300

File: src/components/application_manager/test/request_controller/request_info_test.cc (hunk 3 above)
Comment (VVeremjova) [inline src/components/application_manager/test/request_controller/request_info_test.cc]: @okozlovlux See d960300

File: src/components/application_manager/test/request_controller/request_info_test.cc (hunk 4 above)
Comment (VVeremjova) [inline src/components/application_manager/test/request_controller/request_info_test.cc]: @LevchenkoS See  c94513d

**Comment by okozlovlux** [comment]
@VVeremjova , reviewed

**Comment by nk0leg** [comment]
@VVeremjova, reviewed.

**Comment by VProdanov** [comment]
@VVeremjova Reviewed

**Comment by OHerasym** [comment]
Reviewed

**Comment by anosach-luxoft** [comment]
@VVeremjova please add at least short description to PR why are you going to change `RequestInfo` for UT

**Comment by VVeremjova** [comment]
@anosach-luxoft Done.

**Pull request merged**